### PR TITLE
Add lint-staged, husky for formatting during commits

### LINKED ISSUE #221: Add lint-staged and husky
See [orcahome](https://github.com/orcasound/orcahome) for sample setup:
- https://github.com/orcasound/orcahome/pull/25
- https://github.com/orcasound/orcahome/blob/main/lint-staged.config.js
- https://github.com/orcasound/orcahome/tree/main/.husky

### PR DESCRIPTION
**Pull request by skanderm**
- Closes #221

### CODE REVIEW

**Review by paulcretu — DISMISSED** [review]
Hm not working for me:
```
>>> git commit -m "test"                                                                                                           🐙  ±:test:.
✔ Preparing lint-staged...
⚠ Running tasks for staged files...
  ❯ ui/lint-staged.config.js — 1 file
    ❯ *.{js,jsx,ts,tsx} — 1 file
      ✖ next lint --fix --file ui/src/pages/index.tsx [ENOENT]
    ❯ * — 1 file
      ✖ prettier --ignore-unknown --write [KILLED]
↓ Skipped because of errors from tasks.
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ next lint --fix --file ui/src/pages/index.tsx failed without output (ENOENT).

✖ prettier --ignore-unknown --write failed without output (KILLED).
husky - pre-commit script failed (code 1)
```

File: ui/lint-staged.config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const path = require('path')
+
+const buildEslintCommand = (filenames) =>
+  `next lint --fix --file ${filenames
+    .map((f) => path.relative(process.cwd(), f))
+    .join(' --file ')}`
+
+module.exports = {
+  '*.{js,jsx,ts,tsx}': [buildEslintCommand], // eslint
+  '*': 'prettier --ignore-unknown --write', // prettier
```
Comment (paulcretu) [inline ui/lint-staged.config.js:12]: I think there's a potential race condition here: https://github.com/lint-staged/lint-staged#task-concurrency

File: ui/lint-staged.config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// lint-staged configuration for next 12
+// from https://nextjs.org/docs/basic-features/eslint#lint-staged
```
[the remark this answers is not in the document]
Comment (skanderm) [reply ui/lint-staged.config.js:2]: I'll update this

File: ui/lint-staged.config.js (hunk 1 above)
Comment (skanderm) [reply ui/lint-staged.config.js]: Alrighty, turned off concurrency

**Pull request merged**